apiserver/facades/client/highavailability/highavailability.go

```diff
@@ -167,27 +149,52 @@ func (api *HighAvailabilityAPI) enableHASingle(st *state.State, spec params.Cont
 	if err != nil {
 		return params.ControllersChanges{}, errors.Annotate(err, "retrieving controller config")
 	}
+	if err = validateCurrentControllers(st, cfg, cInfo.MachineIds); err != nil {
+		return params.ControllersChanges{}, errors.Trace(err)
+	}
 	spec.Constraints.Spaces = cfg.AsSpaceConstraints(spec.Constraints.Spaces)
 
-	changes, err := st.EnableHA(spec.NumControllers, spec.Constraints, series, spec.Placement, api.machineID)
+	// Might be nicer to pass the spec itself to this method.
+	changes, err := st.EnableHA(spec.NumControllers, spec.Constraints, spec.Series, spec.Placement, api.machineID)
 	if err != nil {
 		return params.ControllersChanges{}, err
 	}
 	return controllersChanges(changes), nil
 }
 
-// getBootstrapConstraints attempts to return the constraints for the initial
-// bootstrapped controller.
-func getBootstrapConstraints(st *state.State) (constraints.Value, error) {
-	controllerInfo, err := st.ControllerInfo()
-	if err != nil {
-		return constraints.Value{}, err
+// validateCurrentControllers checks for a scenario where there is no HA space
+// in controller configuration and more than one machine-local address on any
+// of the controller machines. An error is returned if it is detected.
+func validateCurrentControllers(st *state.State, cfg controller.Config, machineIds []string) error {
+	if cfg.JujuHASpace() != "" {
+		return nil
+	}
+
+	var badIds []string
+	for _, id := range machineIds {
+		controller, err := st.Machine(id)
+		if err != nil {
+			return errors.Annotatef(err, "reading controller id %v", id)
+		}
+		internal, ok := network.SelectInternalAddresses(controller.Addresses(), false)
+		if !ok || len(internal) > 1 {
+			badIds = append(badIds, id)
+		}
 	}
+	if len(badIds) > 0 {
+		return errors.Errorf(
+			"juju-ha-space is not set and a unique cloud-local address was not found for machines: %v", badIds)
```

Comment:
I wonder about the formatting of this. I think its ok, but it is nice to give users nice looking errors.
Something like:
strings.Join(badIds, ", ")
might be easier to read.
I think it would print:

found for machines: 0, 1, 2
instead of
found for machines: ["0" "1" "2"]